src/unwrap.jl

```diff
@@ -0,0 +1,373 @@
+module Unwrap
+
+import Compat: selectdim
+import Compat
+
+export unwrap, unwrap!
+
+"""
+    unwrap!(m; dims=nothing, range=2pi)
+
+In-place version of [`unwrap`](@ref).
+"""
+function unwrap!(m::AbstractArray{T,N}; dims=nothing, range=2T(pi), kwargs...) where {T, N}
+    if dims === nothing && N != 1
+        Base.depwarn("`unwrap!(m::AbstractArray)` is deprecated, use `unwrap!(m, dims=ndims(m))` instead", :unwrap!)
+        dims = N
+    end
+    unwrap!(m, m; dims=dims, range=range, kwargs...)
+end
+
+"""
+    unwrap!(y, m; dims=nothing, range=2pi)
+
+Unwrap `m` storing the result in `y`, see [`unwrap`](@ref).
+"""
+function unwrap!(y::AbstractArray{T,N}, m::AbstractArray{T,N}; dims=nothing, range::Number=2T(pi), kwargs...) where {T, N}
+    if dims === nothing
+        if N != 1
+            throw(ArgumentError("`unwrap!`: required keyword parameter dims missing"))
+        end
+        dims = 1
+    end
+    if dims isa Integer
+        Compat.accumulate!(unwrap_kernel(range), y, m, dims=dims)
+    elseif dims == 1:N
+        unwrap_nd!(y, m; range=range, kwargs...)
+    else
+        throw(ArgumentError("`unwrap!`: Invalid dims specified: $dims"))
+    end
+    return y
+end
+
+unwrap_kernel(range=2π) = (x, y) -> y - round((y - x) / range) * range
+
+"""
+    unwrap(m; dims=nothing, range=2pi)
+
+Assumes `m` (along the single dimension `dims`) to be sequences of values that
+have been wrapped to be inside the given `range` (centered around
+zero), and undoes the wrapping by identifying discontinuities.
+
+A common usage is for a phase measurement over time, such as when
+comparing successive frames of a short-time-fourier-transform, as
+each frame is wrapped to stay within (-pi, pi].
+
+If `dims` is a `Range`, then the data is unwrapped according to
+the number of dimensions passed to `dims`.
+
+# Arguments
+- `m::AbstractArray{T, N}`: Array to unwrap
+- `range=2pi`: Range of wrapped array
+- `wrap_around=`:  When an element of this tuple is `true`, the
+    unwrapping process will consider the edges along the corresponding axis
+    of the image to be connected
+- `seed=-1`: Unwrapping of arrays with dimension > 1 uses a random initialization. This
```

Comment:
Why pass the seed, if you can just manually seed the RNG before calling `unwrap`? Alternatively, one might wish to pass the RNG.